scripts/release.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/sh
+
+# check if `yq` tool is installed.
+if ! command -v yq &> /dev/null
+then
+	echo "yq tool is required, but could not be found."
+	echo "Install it via: $ brew install yq"
+	exit 1
+fi
+
+# check if `gh` tool is installed.
+if ! command -v gh &> /dev/null
+then
+	echo "Github CLI tool is required, but could not be found."
+	echo "Install it via: $ brew install gh"
+	exit 1
+fi
+
+# check if `gh` tool has auth access.
+# command will return non-zero if not auth'd.
+authd=$(gh auth status -t)
+if [[ $? != 0 ]]
+then
+	echo "ex: $ gh auth login"
+	exit 1
+fi
+
+# check that we're on the `main` branch
+branch=$(git rev-parse --abbrev-ref HEAD)
+if [ $branch != 'main' ]
+then
+	echo "The 'main' must be the current branch to make a release."
+	echo "You are currently on: $branch"
+	exit 1
+fi
+
+if [ -n "$(git status --porcelain)" ]
+then
+  echo "There are uncommited changes. Please commit and create a pull request or stash them.";
+  exit 1
+fi
+
+version=$(yq eval .version pubspec.yaml)
+echo "Appcues plugin package current version: $version"
+
+# no args, so give usage.
+if [ $# -eq 0 ]
+then
+	echo "Release automation script"
+	echo ""
+	echo "Usage: $ ./release.sh <version>"
+	echo "   ex: $ ./release.sh \"1.0.2\""
+	exit 0
+fi
+
+newVersion="${1}"
+echo "Preparing to release $newVersion..."
+
+versionComparison=$(./scripts/semver.sh $newVersion $version)
+
+if [ $versionComparison != '1' ]
+then
+	echo "New version must be greater than previous version ($version)."
+	exit 1
+fi
+
+read -r -p "Are you sure you want to release $newVersion? [y/N] " response
+case "$response" in
+	[yY][eE][sS]|[yY])
+		;;
+	*)
+		exit 1
+		;;
+esac
+
+# generate the changelog for the git release update
+releaseChangelog=$(git log --pretty=format:"- [%as] %s (%h)" $(git describe --tags --abbrev=0 @^)..@ --abbrev=7 | sed '/[🔧🎬⬆️📸✅💡📝]/d')
+releaseTempFile=$(mktemp)
+echo "$releaseChangelog" >> $releaseTempFile
+
+# generate the updated CHANGELOG.md for the repo (Flutter requirement)
+changelog="## $newVersion\n"
+changelog+=$(git log --pretty=format:"* %s (%h)" $(git describe --tags --abbrev=0 @^)..@ --abbrev=7 | sed '/[🔧🎬⬆️📸✅💡📝]/d')
+changelog+="\n"
+changelogTempFile=$(mktemp)
+echo "$changelog" >> $changelogTempFile
+cat CHANGELOG.md >> $changelogTempFile
+mv $changelogTempFile CHANGELOG.md
+
+# update pubspec.yaml version.
+yq -i e '.version = "'${newVersion}'"' pubspec.yaml
+
+# update example/pubspec.yaml version.
+yq -i e '.version = "'${newVersion}'"' example/pubspec.yaml
+
+# ios/appcues_flutter.podspec - version
+sed -i '' -e "s/$version/$newVersion/g" ios/appcues_flutter.podspec
+
+# android/build.gradle - version
+sed -i '' -e "s/$version/$newVersion/g" android/build.gradle
```

Comment:
I'm not 100% sure why you have to separately declare and update both of these and they don't just inherit from the pubspec.yaml version, but I've yet to find any way to do that.  This was how the project template set it up.